diagram-viewer/src/main/java/com/powsybl/diagram/viewer/MainViewController.java

```diff
@@ -43,6 +43,75 @@
  */
 public class MainViewController {
 
+    public enum ComponentFilterType {
+        ALL,
+        HVDC_LINE,
+        BUS,
+        SWITCH,
+        BUSBAR_SECTION,
+        LINE,
+        TIE_LINE,
+        TWO_WINDINGS_TRANSFORMER,
+        THREE_WINDINGS_TRANSFORMER,
+        GENERATOR,
+        BATTERY,
+        LOAD,
+        SHUNT_COMPENSATOR,
+        DANGLING_LINE,
+        STATIC_VAR_COMPENSATOR,
+        LCC_CONVERTER_STATION,
+        VSC_CONVERTER_STATION,
+        GROUND;
+
+        IdentifiableType toIidm() {
+            return switch (this) {
+                case ALL -> null;
+                case HVDC_LINE -> IdentifiableType.HVDC_LINE;
+                case BUS -> IdentifiableType.BUS;
+                case SWITCH -> IdentifiableType.SWITCH;
+                case BUSBAR_SECTION -> IdentifiableType.BUSBAR_SECTION;
+                case LINE -> IdentifiableType.LINE;
+                case TIE_LINE -> IdentifiableType.TIE_LINE;
+                case TWO_WINDINGS_TRANSFORMER -> IdentifiableType.TWO_WINDINGS_TRANSFORMER;
+                case THREE_WINDINGS_TRANSFORMER -> IdentifiableType.THREE_WINDINGS_TRANSFORMER;
+                case GENERATOR -> IdentifiableType.GENERATOR;
+                case BATTERY -> IdentifiableType.BATTERY;
+                case LOAD -> IdentifiableType.LOAD;
+                case SHUNT_COMPENSATOR -> IdentifiableType.SHUNT_COMPENSATOR;
+                case DANGLING_LINE -> IdentifiableType.DANGLING_LINE;
+                case STATIC_VAR_COMPENSATOR -> IdentifiableType.STATIC_VAR_COMPENSATOR;
+                case VSC_CONVERTER_STATION, LCC_CONVERTER_STATION -> IdentifiableType.HVDC_CONVERTER_STATION;
+                case GROUND -> IdentifiableType.GROUND;
+            };
+        }
+
+        ComponentFilterType() { /* compiled code */ }
+
+        @Override
+        public String toString() {
+            return switch (this) {
+                case ALL -> "All";
+                case HVDC_LINE -> "HVDC line";
+                case BUS -> "Bus";
+                case SWITCH -> "Switch";
+                case BUSBAR_SECTION -> "Busbar section";
+                case LINE -> "Line";
+                case TIE_LINE -> "Tie line";
+                case TWO_WINDINGS_TRANSFORMER -> " Two winding transformer";
+                case THREE_WINDINGS_TRANSFORMER -> "Three winding transformer";
```

Comment:
nice, you didn't put the 's', which is an old powsybl mistake! but you should put the dash though
```suggestion
                case TWO_WINDINGS_TRANSFORMER -> " Two-winding transformer";
                case THREE_WINDINGS_TRANSFORMER -> "Three-winding transformer";
```